src/main/java/net/neoforged/neoforge/network/registration/NetworkRegistry.java

```diff
@@ -719,6 +704,39 @@ public static Set<ResourceLocation> getInitialServerUnregisterChannels() {
         return nowForgottenChannels.build();
     }
 
+    /**
+     * Replaces any existing common channels with the incoming ones from a {@link CommonRegisterPayload}.
```

Comment:
Is that the specified behavior of `c:register` on Fabric -- it _replaces_ all registrations with the registrations specified, instead of adding the new registrations? I would think it mimics `minecraft:register`, but I'd like to be sure just in case.

(@Technici4n, could you please advise on this?)

Reply:
I wasn't sure, and didn't see it formally specified anywhere.  Having the behavior of `minecraft:register` would be weird because there isn't a `c:unregister` payload to complement that behavior.

Reply:
We could check the implementation of the packet in Fabric API (and/or consult with @fabricmc maintainers) to see if it does a replacement or a simple addition.

Reply:
> ... there isn't a c:unregister payload ...

there is a `minecraft:unregister` too

Reply:
The `minecraft:` payloads do not interact with the channels supplied by the `c:` payload.  Thus I think the best implementation we can have is for `c:register` to replace the existing channels when it comes in, with the expectation that it be sent one time during configuration.